File: ListFactors.vb
Public Class ListFactors

    Dim find_my_factors As Integer
    Dim found_factors As New List(Of Integer)
    Dim sqrt_range As Integer

    ' Simulate a constructor
    Public Sub _init_(candidate As Integer)
        find_my_factors = candidate
        found_factors.Add(1)
        found_factors.Add(find_my_factors) ' 1 and itself are automatic factors
        sqrt_range = Math.Ceiling(Math.Sqrt(find_my_factors))
    End Sub

    ' Returns an array reference Of the desired factors
    Public Function findFactors() As List(Of Integer)
        ' Loop through 1 To 'find_my_factors' and test for divisibility.
        For count = 2 To sqrt_range
            If find_my_factors Mod count = 0 Then
                found_factors.Add(count)
                ' Get the complementing factor by dividing 'find_my_factor' by variable count.
                found_factors.Add(find_my_factors / count)
            End If
        Next count

        ' Sort the array in ascending order Not entirely necessary.
        found_factors.Sort()

        Return found_factors

    End Function

End Class

File: ListFactors.py
from math import ceil, sqrt

# A class
class Factors:
    # Simulate a constructor
    def __init__(self, candidate):
        self.find_my_factors = candidate
        self.found_factors = [1, self.find_my_factors] # 1 and itself are automatic factors
        self.sqrt_range = ceil(sqrt(self.find_my_factors))

    # Returns an array reference of the desired factors
    def findFactors(self):
        # Loop through 1 to 'find_my_factors' and test for divisibility.
        for self.count in range(2, self.sqrt_range):
            if self.find_my_factors % self.count == 0:
                self.found_factors.append(self.count)
                # Get the complementing factor by dividing 'find_my_factor' by variable count.
                self.found_factors.append(int(self.find_my_factors / self.count))

        # Sort the array in ascending order Not entirely necessary.
        self.found_factors.sort()

        return self.found_factors

